R/CreatePackageReport.R

```diff
@@ -1,17 +1,20 @@
-#' @title Creates a 
+#' @title Surface the internal and external dependecies of an R package.
 #' @name CreatePackageReport
-#' @description Obtain Ratio of Coverage For Each Function Within A Package
+#' @description Surface the internal and external dependecies of an R package. 
 #' @author B. Burns
 #' @param packageName name of a package
 #' @param packageReporters a list of package reporters
+#' @param packagePath (optional) the path to the package repository.  
+#' If given, coverage will be calculated for each function.
 #' @importFrom assertthat assert_that is.string
 #' @importFrom covr package_coverage tally_coverage
 #' @importFrom data.table as.data.table setnames
 #' @importFrom methods is
 #' @return A list of instantiated packageReporters the user can then interact with
 #' @export
 CreatePackageReport <- function(packageName
-                                , packageReporters = DefaultReporters()) {
+                                , packageReporters = DefaultReporters()
+                                , packagePath = NULL) {
```

Comment:
I think that instead of sewing this throughout `CreatePackageReport`, we should create a `PackageCoverageReporter` that gets initialized with this `packagePath` and implements `package_test_coverage`.

That's a lot cleaner than exposing this argument and passing it around just for this one operation. To allow people to color nodes by coverage, we could support some more general way to join in data (like package coverage)

I'd like @patrick-boueri to weigh in on this as well

Reply:
I think this should be the defacto interface and single entry point. I agree we shouldn't overload it with args, but i don't see this going wild beyond this. in principle if we add a ton more, then yeah it should be encapsulated and passed as an object, but i do think it's really useful to have a single function call that does a lot of sane sensible things

Reply:
sure fair enough